src/utils/crypto.util.ts

```diff
@@ -0,0 +1,53 @@
+import { NativeModules } from 'react-native';
+import { BinaryLike } from 'crypto';
+
+interface EncryptedData {
+  cipher: string;
+  iv: string;
+}
+
+interface EncryptedDataSalt {
+  salt: string;
+}
+
+export const generateSalt = (byteCount = 32): string => {
+  const view = new Uint8Array(byteCount);
+
+  crypto.getRandomValues(view);
+
+  // TODO: fix ts (getRandomValues mutates array)
+  // @ts-ignore
+  return btoa(String.fromCharCode.apply(null, view));
+};
+
+export const generateKey = (password: BinaryLike, salt: BinaryLike): Promise<string> =>
+  NativeModules.Aes.pbkdf2(password, salt, 5000, 256);
+
+export const encryptString = async (str: string, password: string): Promise<EncryptedData & EncryptedDataSalt> => {
+  const salt = generateSalt();
+  const key = await generateKey(password, salt);
+
+  const iv: string = await NativeModules.Aes.randomKey(16);
+  const result: EncryptedData = await NativeModules.Aes.encrypt(str, key, iv).then((cipher: string) => ({
+    cipher,
+    iv
+  }));
+
+  return { ...result, salt };
+};
+
+export const encryptObject = async (obj: Record<string, string>, password: string) =>
+  Promise.all(Object.keys(obj).map(async key => ({ [key]: await encryptString(obj[key], password) }))).then(arr =>
+    arr.reduce<Record<string, EncryptedData & EncryptedDataSalt>>((prev, curr) => ({ ...prev, ...curr }), {})
+  );
```

Comment:
This can be done with Object.entries and Object.fromEntries. If React Native supports them!